From e72265be9d8fd0fe023bc5074edd7b7f26f9fc2b Mon Sep 17 00:00:00 2001
From: Ken Williams <kenahoo@gmail.com>
Date: Wed, 18 Jan 2012 19:25:52 -0800
Subject: [PATCH 1/4] Need to set image_title variable when calling by node ID too

---
 image_title.module |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/image_title.module b/image_title.module
index 477d08f..db53be7 100644
--- a/image_title.module
+++ b/image_title.module
@@ -367,6 +367,7 @@ function image_title_preprocess_page(&$variables) {
       $image = theme('image', $thisnode->image_title, $thisnode->title, $thisnode->title, '', FALSE);
       $title = '<span class="txt_title">' . $thisnode->title . '</span>' . $image;
       $variables['title'] = $title;
+      $variables['image_title'] = $title;
     }
   } 
   else {
-- 
1.7.2.5

