From 19e98968bc0099946970f8ddf7423882ee382d47 Mon Sep 17 00:00:00 2001
From: Ken Williams <kenahoo@gmail.com>
Date: Wed, 18 Jan 2012 20:27:42 -0800
Subject: [PATCH 2/4] Factor out some common code to the two cases in image_title_preprocess_page()

---
 image_title.module |   38 +++++++++++++++++++++++---------------
 1 files changed, 23 insertions(+), 15 deletions(-)

diff --git a/image_title.module b/image_title.module
index db53be7..3a70e7a 100644
--- a/image_title.module
+++ b/image_title.module
@@ -363,11 +363,11 @@ function image_title_preprocess_page(&$variables) {
   
   if (arg(0) === 'node' && is_numeric(arg(1))) {
     $thisnode = node_load($variables['node']->nid);
-    if ($thisnode->image_title && $thisnode->image_title_status ) { 
-      $image = theme('image', $thisnode->image_title, $thisnode->title, $thisnode->title, '', FALSE);
-      $title = '<span class="txt_title">' . $thisnode->title . '</span>' . $image;
-      $variables['title'] = $title;
-      $variables['image_title'] = $title;
+    if ($thisnode->image_title && $thisnode->image_title_status ) {
+      $orig_title = $thisnode->title;
+      $im_path = $thisnode->image_title;
+    } else {
+      return;
     }
   } 
   else {
@@ -378,17 +378,25 @@ function image_title_preprocess_page(&$variables) {
     //load menu item page data
     $item = menu_get_item();
     //check for an image for this page
-    if ($image_title = db_fetch_array(db_query('SELECT image FROM {image_title_menu} WHERE path = "%s"', $item['path']))) {
-      $image = theme('image', $image_title['image'], $item['title'], $item['title'], '', FALSE);
-      $title = '<span class="txt_title">' . $item['title'] . '</span>' . $image;
-      $variables['title'] = $title;
-      //we set another variable here because some modules may override $title
-      //later on, so you can in turn override their values in your theme
-      //template.php file with an appropriate preprocess containing simply:
-      //  $variables['title'] = $variables['image_title'];
-      $variables['image_title'] = $title;
+    if ($image_title = db_fetch_array(
+        db_query('SELECT image FROM {image_title_menu} WHERE path = "%s"', $item['path'])
+       )) {
+      $orig_title = $item['title'];
+      $im_path = $image_title['image'];
+    } else {
+      return;
     }
-  }  
+  }
+
+  //we set another variable here because some modules may override $title
+  //later on, so you can in turn override their values in your theme
+  //template.php file with an appropriate preprocess containing simply:
+  //  $variables['title'] = $variables['image_title'];
+
+  $image = theme('image', $im_path, $orig_title, $orig_title, '', FALSE);
+  $title = '<span class="txt_title">' . $orig_title . '</span>' . $image;
+  $variables['title'] = $title;
+  $variables['image_title'] = $title;  
 }
 
 /**
-- 
1.7.2.5

